x-pack/elastic-agent/pkg/agent/storage/storage_test.go

```diff
@@ -49,6 +49,11 @@ func TestReplaceOrRollbackStore(t *testing.T) {
 
 		require.True(t, bytes.Equal(writtenContent, replaceWith))
 		requireFilesCount(t, dir, 2)
+
+		info, err := os.Stat(target)
+		require.NoError(t, err)
+
+		require.Equal(t, perms, info.Mode())
```

Comment:
please include windows fix in this cherry pick so we dont break 7.x build process